omod/src/main/webapp/pages/login.gsp

```diff
@@ -21,6 +21,15 @@
     var OPENMRS_CONTEXT_PATH = '${ ui.contextPath() }';
 </script>
 
+<script type="text/javascript">
+	jQuery(function () {
+    	jQuery("#togglePassword").click(function () {
+        	jQuery(this).toggleClass("fa-eye fa-eye-slash");
+            var type = jQuery(this).hasClass("fa-eye-slash") ? "text" : "password";
+            jQuery("#password").attr("type", type);
+        });
```

Comment:
hey @ibacher 
I think the symbol **$** wasn't defined 
![image](https://user-images.githubusercontent.com/58003327/126450803-d223cfb2-4977-4cdf-8052-a1b35d4c5bfe.png)

so I have tried to change the variable name from 

> var $this = jQuery(this);

`var pwdIcon = jQuery(this);`

Reply:
Oh... right. This is in a JSP so that `$this` gets read by Groovy and tries to interpolate the value, then it doesn't work because it would need to be, e.g., `${this}` and then it's whatever the `toString()` method of the Groovy page returns, so probably not what we need. 

Good job renaming the variable. I was just trying to replicate the thing that's sometimes done of prefixing a variable with `$` (usually how you can reference jQuery) to indicate it was already a jQuery object.

Reply:
thanks for the great explanation @ibacher 
I think this is now almost good.  cc @isears